relay-profiling/src/transaction_metadata.rs

```diff
@@ -2,33 +2,52 @@ use serde::{Deserialize, Serialize};
 
 use relay_general::protocol::EventId;
 
-use crate::utils::deserialize_number_from_string;
+use crate::utils::{deserialize_number_from_string, is_zero};
 
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct TransactionMetadata {
     pub id: EventId,
     pub name: String,
     pub trace_id: EventId,
 
+    #[serde(
+        default,
+        deserialize_with = "deserialize_number_from_string",
+        skip_serializing_if = "is_zero"
+    )]
+    pub active_thread_id: u64,
+
     #[serde(deserialize_with = "deserialize_number_from_string")]
     pub relative_start_ns: u64,
     #[serde(deserialize_with = "deserialize_number_from_string")]
     pub relative_end_ns: u64,
 
     // Android might have a CPU clock for the trace
-    #[serde(default, deserialize_with = "deserialize_number_from_string")]
+    #[serde(
+        default,
+        deserialize_with = "deserialize_number_from_string",
+        skip_serializing_if = "is_zero"
+    )]
     pub relative_cpu_start_ms: u64,
-    #[serde(default, deserialize_with = "deserialize_number_from_string")]
+    #[serde(
+        default,
+        deserialize_with = "deserialize_number_from_string",
+        skip_serializing_if = "is_zero"
+    )]
     pub relative_cpu_end_ms: u64,
 }
 
 impl TransactionMetadata {
     pub fn valid(&self) -> bool {
         !self.id.is_nil()
-            && !self.trace_id.is_nil()
             && !self.name.is_empty()
-            && self.relative_start_ns < self.relative_end_ns
+            && !self.trace_id.is_nil()
             && self.relative_cpu_start_ms <= self.relative_cpu_end_ms
+            && self.relative_start_ns < self.relative_end_ns
```

Comment:
Why does `relative_start_ns` use `<` but `relative_cpu_start_ms` use `<=`? Are 0 duration profiles allowed when using `relative_cpu_start_ms` (this is android only right?)?

Reply:
It's to avoid considering the profile invalid when we don't have a value (fields will be defaulted to 0). Ideally we'd do a check based on the platform but we don't have access to that. We might also have to align both to check for `<=` because we might have CPU time only anyway.